tasmota/xdrv_01_webserver.ino

```diff
@@ -2114,36 +2111,46 @@ void HandleWifiConfiguration(void)
           }
         }
 
-        // remove duplicates ( must be RSSI sorted )
-        String cssid;
-        for (uint32_t i = 0; i < n; i++) {
-          if (-1 == indices[i]) { continue; }
-          cssid = WiFi.SSID(indices[i]);
-          uint32_t cschn = WiFi.channel(indices[i]);
-          for (uint32_t j = i + 1; j < n; j++) {
-            if ((cssid == WiFi.SSID(indices[j])) && (cschn == WiFi.channel(indices[j]))) {
-              DEBUG_CORE_LOG(PSTR(D_LOG_WIFI D_DUPLICATE_ACCESSPOINT " %s"), WiFi.SSID(indices[j]).c_str());
-              indices[j] = -1;  // set dup aps to index -1
-            }
-          }
-        }
-
         //display networks in page
-        for (uint32_t i = 0; i < n; i++) {
-          if (-1 == indices[i]) { continue; }  // skip dups
+        uint32_t i = 0;
+        while (i < n) {
           int32_t rssi = WiFi.RSSI(indices[i]);
+          String ssid = WiFi.SSID(indices[i]);
           DEBUG_CORE_LOG(PSTR(D_LOG_WIFI D_SSID " %s, " D_BSSID " %s, " D_CHANNEL " %d, " D_RSSI " %d"),
-            WiFi.SSID(indices[i]).c_str(), WiFi.BSSIDstr(indices[i]).c_str(), WiFi.channel(indices[i]), rssi);
-          int quality = WifiGetRssiAsQuality(rssi);
-          int auth = WiFi.encryptionType(indices[i]);
-          char encryption[20];
-          WSContentSend_P(PSTR("<div><a href='#p' onclick='c(this)'>%s</a>&nbsp;(%d)&nbsp<span class='q'>%s %d%% (%d dBm)</span></div>"),
-            HtmlEscape(WiFi.SSID(indices[i])).c_str(),
-            WiFi.channel(indices[i]),
-            GetTextIndexed(encryption, sizeof(encryption), auth +1, kEncryptionType),
-            quality, rssi
+            ssid.c_str(), WiFi.BSSIDstr(indices[i]).c_str(), WiFi.channel(indices[i]), rssi);
+
+          // Print SSID
+          WSContentSend_P(PSTR("<div><a href='#p' onclick='c(this)'>%s</a><br><ul>"),
+            HtmlEscape(ssid).c_str()
           );
-          delay(0);
+
+          int j = 0;
+          String nextSSID = "";
```

Comment:
You can save code with
`String nextSSID;`

You don't need to give it a value, is it still initialized.